src/scriptdev2/scripts/kalimdor/durotar.cpp

```diff
@@ -33,89 +33,184 @@ EndContentData */
+    uint32 m_uiChopTimer; //Time, until the npc stops chopping at the tree
+    uint32 m_uiKneelTimer; //Time, delay kneeling to allow sound to finish playing
+    uint32 m_uiGetUpTimer; //Time, npc stays at the pile so kneel animation can finish
+    bool m_bIsAtPile;
+    float m_fSpawnO;
+
+    void RestartWakeTimer() {
+        m_uiWakeTimer = urand(100000, 120000); //Random wakeup timer 1m40s to 2m, retail times unknown, 2m is duration of sleep aura
+        m_uiSleepTimer = 0; //Ensure only one of the two timers is running at any given time
+    }
+
+    void RestartSleepTimer() {
+        m_uiSleepTimer = 80000; //Go to sleep after 1m20s, confirmed blizzlike
+        m_uiWakeTimer = 0; //Ensure only one of the two timers is running at any given time
+    }
 
     void Reset() override
     {
-        m_uiResetSleepTimer = 0;
-        m_uiStopSleepingTimer = urand(90000, 120000);       // Sleeping aura has only 2min duration
+        RestartWakeTimer();
+        m_uiGetUpTimer = 0;
+        m_uiKneelTimer = 0;
+        m_uiChopTimer = 0;
+        m_bIsAtPile = false;
     }
 
-    // Can also be self invoked for random working
-    void StartLumbering(Unit* pInvoker)
+    void Awaken(Unit* pInvoker)
     {
-        m_uiStopSleepingTimer = 0;
-        if (GameObject* pLumber = GetClosestGameObjectWithEntry(m_creature, GO_LUMBERPILE, 15.0f))
+        if (pInvoker->GetTypeId() == TYPEID_PLAYER)
         {
-            m_creature->RemoveAurasDueToSpell(SPELL_PEON_SLEEP);
+            DoScriptText(SAY_PEON_AWOKEN, m_creature, pInvoker);
+            m_creature->PlayDistanceSound(urand(0, 1) ? SOUND_PEON_AWAKE_1 : SOUND_PEON_AWAKE_2);
+            m_creature->SetWalk(false); //peon will run if awoken by player
+            ((Player*)pInvoker)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
+        }
+        else
+            m_creature->SetWalk(true); //peon will walk if awoken on its own
+
+        m_creature->RemoveAurasDueToSpell(SPELL_PEON_SLEEP);
+        RestartSleepTimer();
+        GoLumberPile();
+    }
 
+    void GoLumberPile()
+    {
+        if (GameObject* pLumber = GetClosestGameObjectWithEntry(m_creature, GO_LUMBERPILE, 15.0f))
+        {
             float fX, fY, fZ;
-            m_creature->SetWalk(false);
             pLumber->GetContactPoint(m_creature, fX, fY, fZ, CONTACT_DISTANCE);
-
-            if (pInvoker->GetTypeId() == TYPEID_PLAYER)
-            {
-                DoScriptText(SAY_PEON_AWAKE_1, m_creature);
-                ((Player*)pInvoker)->KilledMonsterCredit(m_creature->GetEntry(), m_creature->GetObjectGuid());
-                m_creature->GetMotionMaster()->MovePoint(1, fX, fY, fZ);
-            }
-            else
-                m_creature->GetMotionMaster()->MovePoint(2, fX, fY, fZ);
+            m_creature->HandleEmote(EMOTE_STATE_NONE); //cancel chopping
+            m_creature->GetMotionMaster()->MovePoint(1, fX, fY, fZ);
```

Comment:
@xfurry
We are missing serverside tree GO waypoints; https://github.com/cmangos/issues/issues/1143#issuecomment-260153214 - if we had these we wouldn't need to use the waypoint system at all and could do it all through the point movement system;
> ```code
ServerToClient: SMSG_ON_MONSTER_MOVE (0x6E17)
GUID: Full: 0xF130293C000065D4 Type: Creature Entry: 10556 Low: 26068
Spline Type: FacingTarget (3)
Facing GUID: Full: 0xF112AEA700002B09 Type: GameObject Entry: 175783 Low: 11017
```
Object missing in WoTLKDB... and probably Classic/TBC also... (i'll add soon*)
```code
175783: TreeWayPoint (DND)
Type: 5
displayID: 327
Size: 1
Data1: 1
Data2: 1
Data6: -1
```
Unfortunately - spawnpoints for these objects are hidden